src/pages/sc-cycle-counts/Forms/CycleCountsForm.js

```diff
@@ -0,0 +1,465 @@
+            return !!value
+          }
+
+          return true
+        }),
+    }),
+    onSubmit: async obj => {
+      try {
+        const response = await postRequest({
+          extension: SCRepository.StockCount.set,
+          record: JSON.stringify(obj)
+        })
+
+        if (!obj.recordId) {
+          setStore(prevStore => ({
+            ...prevStore,
+            recordId: response.recordId
+          }))
+          toast.success(platformLabels.Added)
+          formik.setFieldValue('recordId', response.recordId)
+          const res2 = await getData(response.recordId);
+
+          formik.setValues({
+            ...res2.record,
+            date: formatDateFromApi(res2.record.date)
+          })
+        } else toast.success(platformLabels.Edited)
+
+        invalidate()
+      } catch (error) {}
+    }
+  })
+
+  const onWorkFlowClick = async () => {
+    stack({
+      Component: WorkFlow,
+      props: {
+        functionId: SystemFunction.StockCount,
+        recordId: formik.values.recordId
+      },
+      width: 950,
+      title: 'Workflow'
+    })
+  }
+
+  async function getData(recordId) {
+    try {
+      return await getRequest({
+        extension: SCRepository.StockCount.get,
+        parameters: `_recordId=${recordId}`
+      })
+    } catch (error) {}
+  }
+
+  async function refetchForm(recordId) {
+    const res2 = await getData(recordId)
+    res2.record.date = formatDateFromApi(res2.record.date)
+
+    return res2;
+  }
+
+  const isClosed = formik.values.wip === 2
+  const isPosted = formik.values.status === 3
+
+  const onClose = async recId => {
+    try {
+      const res = await postRequest({
+        extension: SCRepository.StockCount.close,
+        record: JSON.stringify({ recordId: recId })
+      })
+
+      toast.success(platformLabels.Closed)
+      invalidate()
+      const res2 = await refetchForm(res.recordId)
+      formik.setValues(res2.record)
+      setStore(prevStore => ({
+        ...prevStore,
+        isClosed: res2.record.wip === 2,
+      }))
+      
+    } catch (error) {}
+  }
+
+  async function onReopen() {
+    try {
+      const res = await postRequest({
+        extension: SCRepository.StockCount.reopen,
+        record: JSON.stringify(formik.values)
+      })
+
+      toast.success(platformLabels.Reopened)
+      invalidate()
+      const res2 = await refetchForm(res.recordId)
+      formik.setValues(res2.record)
+      setStore(prevStore => ({
+        ...prevStore,
+        isClosed: res2.record.wip === 2,
+      }))
+      
+    } catch (error) {}
+  }
+
+  const onPost = async () => {
+    try {
+      const copy = { ...formik.values }
+      copy.date = formatDateToApi(copy.date)
+
+      const res = await postRequest({
+        extension: SCRepository.StockCount.post,
+        record: JSON.stringify(copy)
+      })
+
+      toast.success(platformLabels.Posted)
+      invalidate()
+
+      const res2 = await refetchForm(res.recordId)
```

Comment:
since u are closing the window after post directly then this block serves no purpose
```
 const res2 = await refetchForm(res.recordId)
      formik.setValues(res2.record)
      setStore(prevStore => ({
        ...prevStore,
        isPosted: res2.record.status === 3,
      }))
```